Test.cs

```diff
@@ -12,7 +14,12 @@ public static long calculate(long number)
                 throw new Exception("Number must be non-negative.");
             }
 
-            return number == 0 ? 1 : number * Calculate(number + 1);
+            return number == 0 ? 1 : number * Calculate(number - 1);
+        }
+
```

Comment:
The method name `GetBranchNama` appears to have a typo. It should be corrected to `GetBranchName` for accuracy and consistency. Additionally, the method should return the `branchName` variable, so the code inside the method should be `return branchName;`.